controller/konnect/ops/ops_konnecttransitgateway.go

```diff
@@ -0,0 +1,246 @@
+				Kind:             sdkkonnectcomp.AWSTransitGatewayAttachmentTypeAwsTransitGatewayAttachment,
+				TransitGatewayID: spec.AWSTransitGateway.AttachmentConfig.TransitGatewayID,
+				RAMShareArn:      spec.AWSTransitGateway.AttachmentConfig.RAMShareArn,
+			},
+		}
+	case konnectv1alpha1.TransitGatewayTypeAzureTransitGateway:
+		req.AzureTransitGateway = &sdkkonnectcomp.AzureTransitGateway{
+			Name: spec.AzureTransitGateway.Name,
+			DNSConfig: lo.Map(spec.AWSTransitGateway.DNSConfig, func(dnsConf konnectv1alpha1.TransitGatewayDNSConfig, _ int) sdkkonnectcomp.TransitGatewayDNSConfig {
+				return sdkkonnectcomp.TransitGatewayDNSConfig{
+					RemoteDNSServerIPAddresses: dnsConf.RemoteDNSServerIPAddresses,
+					DomainProxyList:            dnsConf.DomainProxyList,
+				}
+			}),
+			TransitGatewayAttachmentConfig: sdkkonnectcomp.AzureVNETPeeringAttachmentConfig{
+				Kind:              sdkkonnectcomp.AzureVNETPeeringAttachmentTypeAzureVnetPeeringAttachment,
+				TenantID:          spec.AzureTransitGateway.AttachmentConfig.TenantID,
+				SubscriptionID:    spec.AzureTransitGateway.AttachmentConfig.SubscriptionID,
+				ResourceGroupName: spec.AzureTransitGateway.AttachmentConfig.ResourceGroupName,
+				VnetName:          spec.AzureTransitGateway.AttachmentConfig.VnetName,
+			},
+		}
+	}
+
+	return req
+}
+
+func extractKonnectIDFromTransitGatewayResponse(resp *sdkkonnectcomp.TransitGatewayResponse) string {
+	switch resp.Type {
+	case sdkkonnectcomp.TransitGatewayResponseTypeAwsTransitGatewayResponse:
+		return resp.AwsTransitGatewayResponse.ID
+	case sdkkonnectcomp.TransitGatewayResponseTypeAzureTransitGatewayResponse:
+		return resp.AzureTransitGatewayResponse.ID
+	case sdkkonnectcomp.TransitGatewayResponseTypeAwsVpcPeeringGatewayResponse:
+		// AWS VPC peering gateway is not supported yet.
+		return ""
+	}
+	return ""
+}
+
+func extractStateFromTransitGatewayResponse(resp *sdkkonnectcomp.TransitGatewayResponse) sdkkonnectcomp.TransitGatewayState {
+	switch resp.Type {
+	case sdkkonnectcomp.TransitGatewayResponseTypeAwsTransitGatewayResponse:
+		return resp.AwsTransitGatewayResponse.State
+	case sdkkonnectcomp.TransitGatewayResponseTypeAzureTransitGatewayResponse:
+		return resp.AzureTransitGatewayResponse.State
+	case sdkkonnectcomp.TransitGatewayResponseTypeAwsVpcPeeringGatewayResponse:
+		// AWS VPC peering gateway is not supported yet.
+		return sdkkonnectcomp.TransitGatewayState("")
+	}
+	return sdkkonnectcomp.TransitGatewayState("")
+}
+
+func extractNameFromTransitGatewayResponse(resp *sdkkonnectcomp.TransitGatewayResponse) string {
+	switch resp.Type {
+	case sdkkonnectcomp.TransitGatewayResponseTypeAwsTransitGatewayResponse:
+		return resp.AwsTransitGatewayResponse.Name
+	case sdkkonnectcomp.TransitGatewayResponseTypeAzureTransitGatewayResponse:
+		return resp.AzureTransitGatewayResponse.Name
+	case sdkkonnectcomp.TransitGatewayResponseTypeAwsVpcPeeringGatewayResponse:
+		// AWS VPC peering gateway is not supported yet.
```

Comment:
AFAIR this will be caught on the CRD level so I think this is worth mentioning here so that future readers to not need to think: "Is this validated somewhere?"

Reply:
Done.